docs/user_guide/timeseries/forecasting/LagFeatures.rst

```diff
@@ -0,0 +1,257 @@
+.. _lag_features:
+
+.. currentmodule:: feature_engine.timeseries.forecasting
+
+LagFeatures
+===========
+
+The :class:`LagFeatures` adds lag features to the dataframe. A lag feature is a feature
+with information about a prior time step.
+
+In forecasting, past values of the variable we want to forecast, are likely to be
+predictive. Past values of other accompanying predictive features can also help us
```

Comment:
I would probably write this sentence.

In general, the historic data of both the predictive and accompanying features is a useful reference when forecasting. Therefore, it is common practice to create lag features for time-series and predictive variables.

Reply:
the thing is, for this sentence, I borrowed Kishan's words. So let's just merge and then, if he finds it funny he can change them :p